cloudinit/tests/test_features.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+# pylint: disable=no-member,no-name-in-module
+"""
+This file is for testing the feature flag functionality itself,
+NOT for testing any individual feature flag
+"""
+import pytest
+from importlib import reload
+from pathlib import Path
+
+import cloudinit
+import cloudinit.features
+
+
+@pytest.yield_fixture()
+def create_override(request):
+    override_path = Path(cloudinit.__file__).parent / 'feature_overrides.py'
+    with override_path.open('w') as f:
+        for key, value in request.param.items():
+            f.write('{} = {}\n'.format(key, value))
+
+    # The goal is to reload features.py so we also get the
+    # features_overrides.py stuff. This could normally be achieved with
+    # a single reload(cloudinit.features), however, since this fixture will be
+    # called multiple times in the same interpreter, calling
+    # reload(cloudinit.features) a second time won't refresh what already
+    # exists in feature_overrides.py because that's now already defined, so we
+    # need to call reload(cloudinit.feature_overrides) first. But the first
+    # time we call this fixture, cloudinit.feature_overrides doesn't exist yet,
+    # so we need to reload cloudinit.features first before we can reload
+    # cloudinit.feature_overrides, hence the three calls here.
+    reload(cloudinit.features)
+    reload(cloudinit.feature_overrides)
+    reload(cloudinit.features)
+
+    yield
+
+    override_path.unlink()
```

Comment:
I think we also need to reload `features` (and re-remove `feature_overrides` from `sys.modules` beforehand) here, else I _think_ the feature flag overrides from the last test to use this fixture will remain in place for the remainder of the test run.

Reply:
I don't think there's an easy answer to this one. If there are references to a value or it is cached somehow, it will still be around after a module is reloaded, even if it was removed from sys.modules. If the value is _changed_ it will be reloaded as expected. Since this a hack for testing, I don't see much risk here.

Reply:
Yeah, my concern is that if the set state of a feature flag is leaking out then testing that relies on the default value of that feature flag may start failing. However, if that proves to be an issue then we can revisit this then.